src/ChatHistoryItem.tsx

```diff
@@ -32,29 +43,28 @@ export const ChatHistoryItem = memo(
       e.preventDefault();
       e.stopPropagation();
       if (editingTitle) {
+        setEditInputValue(optimisticTitle);
         setEditingTitle(false);
-        setEditInputValue(title);
       }
       setIsMenuOpen((prev) => !prev);
     };
 
-    const handleSaveTitle = useCallback(
-      (currentTitle: string) => {
-        const trimmedTitle = currentTitle.trim();
-        if (trimmedTitle === '') {
-          setEditInputValue(title);
-          setEditingTitle(false);
-          return;
-        }
+    const editInputRef = useRef(editInputValue);
```

Comment:
Using a ref here is more stable - passing in the state value as an arg could cause multiple calls of handleSave, which resulted in a UI flash.